compiler/ccgexprs.nim

```diff
@@ -2345,7 +2345,7 @@ proc genMove(p: BProc; n: PNode; d: var TLoc) =
           else:
             linefmt(p, cpsStmts, "$1($2);$n", [rdLoc(b), byRefLoc(p, a)])
     else:
-      let flags = if not canMove(p, n[1], d): {needToCopy} else: {}
+      let flags = if isSinkType(n[1].typ): {needToCopy} else: {}
```

Comment:
Huh? You cannot move just because it is a sink type and also, why not patch `canMove` instead?

Reply:
In what situation that an object cannot be moved in refc? Before https://github.com/nim-lang/Nim/pull/22229, there is basically no restriction on `move` in refc.